sds_data_manager/lambda_code/IAlirtCode/ialirt_catalog_api.py

```diff
@@ -7,25 +13,83 @@
 logger.setLevel(logging.INFO)
 
 
-def lambda_handler(event, context):
+def lambda_handler(event, context) -> dict:
     """Entry point to the catalog API lambda.
 
-    Blank function for initial setup.
+    Takes in a date range and ground station, and queries the s3 bucket that holds
+    the pointing schedules for ground stations part of the I-ALiRT network.
+
+    If files exist for the requested date range, the response body returns all file
+    versions for every date, including the file name, a pre-signed url for its download,
+    and its last date modified.
 
     Parameters
     ----------
     event : dict
         The JSON formatted document with the data required for the
-        lambda function to process
+        lambda function to process, i.e., date range and ground station
     context : LambdaContext
         This object provides methods and properties that provide
         information about the invocation, function,
         and runtime environment.
+
+    Returns
+    -------
+    dict
+        A dictionary containing headers, status code, and body, designed to be returned
+        as an API response.
     """
-    logger.info(f"Event: {event}")
-    logger.info(f"Context: {context}")
+    logger.info("Received event: " + json.dumps(event, indent=2))
+
+    if not event['start_date'] < event['end_date']:
+        response_body = ("The start date of the requested time frame is not before the "
+                         "end date. Please supply a valid time range.")
+        return {
+            "statusCode": 400,
+            "headers": {"Content-Type": "application/json"},
+            "body": json.dumps(response_body)
+        }
+
+    bucket = os.getenv("S3_BUCKET")
+    region = os.getenv("REGION")
+
+    s3_client = boto3.client(
+        "s3",
+        region_name=region,
+        config=botocore.client.Config(signature_version="s3v4"),
+    )
+
+    response_body = {}
+    for day in np.arange(event['start_date'], event['end_date'], dtype='datetime64[D]'):
+        day_path = (f"pointing_schedules/{event['station']}/" +
+                    day.item().strftime('%Y%m%d') + "/")
+        files = s3_client.list_objects_v2(Bucket=bucket, Prefix=day_path)
+        if 'Contents' not in files.keys():
+            response_body = (
+                f"There are not files associated with the provided date {day}. "
+                f"Please supply a valid time range that is covered by existing files.")
+            return {
+                "statusCode": 404,
+                "headers": {"Content-Type": "application/json"},
+                "body": json.dumps(response_body)
+            }
+        file_names = []
+        file_urls = []
+        dates_modified = []
+        for file in files['Contents']:
+            pre_signed_url = s3_client.generate_presigned_url(
+                "get_object",
+                Params={"Bucket": bucket, "Key": file['Key']}, ExpiresIn=604800
```

Comment:
This is a long expires time, suggest changing it to like 10-15 minutes or so. A user can re-request the data if needed.

Reply:
I'm removing the URL-generation logic of this endpoint, so I won't need to address this comment with a change. 👍